pkg/controller/grouping/group_entity_index_test.go

```diff
@@ -0,0 +1,314 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grouping
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/vmware-tanzu/antrea/pkg/apis/core/v1alpha2"
+	"github.com/vmware-tanzu/antrea/pkg/controller/types"
+)
+
+const (
+	groupType1 = "fakeGroup1"
+	groupType2 = "fakeGroup2"
+)
+
+var (
+	// Fake Pods
```

Comment:
would it be worth having a test with different label names (not just "app")? Or do you think the coverage would be no different?

Reply:
I guess the converage should be same. podBar1 and podFoo1InOtherNamespace are used as noise and to ensure pod label selector and namespace selector are respected. And we are using K8s lib's method to check whether a selector matches a label set, the labels' content is transparent to Antrea.